original patch:
poky rev: 753e2a0ede4449917c75353b57f13bbafe70fac8
file: https://git.yoctoproject.org/cgit/cgit.cgi/poky/tree/meta/recipes-core/sysvinit/sysvinit-2.88dsf/install.patch?id=753e2a0ede4449917c75353b57f13bbafe70fac8

rebased to v2.93 patch:
pull request: https://github.com/masami256/sysvinit/pull/1/

diff --git a/src/Makefile b/src/Makefile
index 3c93783..fd9bb9f 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -77,12 +77,19 @@ else
   INSTALL_DATA	= install -m 644
 endif
 INSTALL_DIR	= install -m 755 -d
-MANDIR		= /usr/share/man
+
+ROOT         ?=
+base_bindir  ?= /bin
+base_sbindir ?= /sbin
+bindir       ?= /usr/bin
+sysconfdir   ?= /etc
+includedir   ?= /usr/include
+mandir       ?= /usr/share/man
 
 ifeq ($(WITH_SELINUX),yes)
   SELINUX_DEF	=  -DWITH_SELINUX
   INITLIBS	+= -lsepol -lselinux
-  SULOGINLIBS	= -lselinux	
+  SULOGINLIBS	= -lselinux
 else
   SELINUX_DEF	=
   INITLIBS	=
@@ -146,14 +153,14 @@ fstab-decode:	LDLIBS += $(STATIC)
 fstab-decode:	fstab-decode.o
 
 sulogin.o:	CPPFLAGS += $(SELINUX_DEF)
-sulogin.o:	sulogin.c 
+sulogin.o:	sulogin.c
 
 runlevellog.o:	runlevellog.h runlevellog.c paths.h
 
 init.o:		CPPFLAGS += $(SELINUX_DEF)
 init.o:		init.c init.h initreq.h paths.h reboot.h runlevellog.h runlevellog.c set.h paths.h
 
-utmp.o:		
+utmp.o:
 
 init_utmp.o:	CPPFLAGS += -DINIT_MAIN
 init_utmp.o:	utmp.c init.h initreq.h paths.h
@@ -181,42 +188,42 @@ clobber:	cleanobjs
 distclean:	clobber
 
 install:	all
-		$(INSTALL_DIR) $(ROOT)/bin/ $(ROOT)/sbin/
-		$(INSTALL_DIR) $(ROOT)/usr/bin/
+		$(INSTALL_DIR) $(ROOT)$(base_bindir)/ $(ROOT)$(base_sbindir)/
+		$(INSTALL_DIR) $(ROOT)$(bindir)/
 		for i in $(BIN); do \
-			$(INSTALL_EXEC) $$i $(ROOT)/bin/ ; \
+			$(INSTALL_EXEC) $$i $(ROOT)$(base_bindir)/ ; \
 		done
 		for i in $(SBIN); do \
-			$(INSTALL_EXEC) $$i $(ROOT)/sbin/ ; \
+			$(INSTALL_EXEC) $$i $(ROOT)$(base_sbindir)/ ; \
 		done
 		for i in $(USRBIN); do \
-			$(INSTALL_EXEC) $$i $(ROOT)/usr/bin/ ; \
+			$(INSTALL_EXEC) $$i $(ROOT)$(bindir)/ ; \
 		done
-		# $(INSTALL_DIR) $(ROOT)/etc/
-		# $(INSTALL_EXEC) ../doc/initscript.sample $(ROOT)/etc/
-		ln -sf halt $(ROOT)/sbin/reboot
-		ln -sf halt $(ROOT)/sbin/poweroff
-		ln -sf init $(ROOT)/sbin/telinit
-		ln -sf /sbin/killall5 $(ROOT)/bin/pidof
-		if [ ! -f $(ROOT)/usr/bin/lastb ]; then \
-			ln -sf last $(ROOT)/usr/bin/lastb; \
+		# $(INSTALL_DIR) $(ROOT)$(sysconfdir)/
+		# $(INSTALL_EXEC) initscript.sample $(ROOT)$(sysconfdir)/
+		ln -sf halt $(ROOT)$(base_sbindir)/reboot
+		ln -sf halt $(ROOT)$(base_sbindir)/poweroff
+		ln -sf init $(ROOT)$(base_sbindir)/telinit
+		ln -sf $(base_sbindir)/killall5 $(ROOT)$(base_bindir)/pidof
+		if [ ! -f $(ROOT)$(bindir)/lastb ]; then \
+			ln -sf last $(ROOT)$(bindir)/lastb; \
 		fi
-		$(INSTALL_DIR) $(ROOT)/usr/include/
-		$(INSTALL_DATA) initreq.h $(ROOT)/usr/include/
-		$(INSTALL_DIR) $(ROOT)$(MANDIR)/man1/
-		$(INSTALL_DIR) $(ROOT)$(MANDIR)/man5/
-		$(INSTALL_DIR) $(ROOT)$(MANDIR)/man8/
+		$(INSTALL_DIR) $(ROOT)$(includedir)/
+		$(INSTALL_DATA) initreq.h $(ROOT)$(includedir)/
+		$(INSTALL_DIR) $(ROOT)$(mandir)/man1/
+		$(INSTALL_DIR) $(ROOT)$(mandir)/man5/
+		$(INSTALL_DIR) $(ROOT)$(mandir)/man8/
 		for man in $(MAN1); do \
-			$(INSTALL_DATA) ../man/$$man $(ROOT)$(MANDIR)/man1/; \
-			sed -i "1{ $(MANDB); }" $(ROOT)$(MANDIR)/man1/$$man ; \
+			$(INSTALL_DATA) ../man/$$man $(ROOT)$(mandir)/man1/; \
+			sed -i "1{ $(MANDB); }" $(ROOT)$(mandir)/man1/$$man ; \
 		done
 		for man in $(MAN5); do \
-			$(INSTALL_DATA) ../man/$$man $(ROOT)$(MANDIR)/man5/; \
-			sed -i "1{ $(MANDB); }" $(ROOT)$(MANDIR)/man5/$$man ; \
+			$(INSTALL_DATA) ../man/$$man $(ROOT)$(mandir)/man5/; \
+			sed -i "1{ $(MANDB); }" $(ROOT)$(mandir)/man5/$$man ; \
 		done
 		for man in $(MAN8); do \
-			$(INSTALL_DATA) ../man/$$man $(ROOT)$(MANDIR)/man8/; \
-			sed -i "1{ $(MANDB); }" $(ROOT)$(MANDIR)/man8/$$man ; \
+			$(INSTALL_DATA) ../man/$$man $(ROOT)$(mandir)/man8/; \
+			sed -i "1{ $(MANDB); }" $(ROOT)$(mandir)/man8/$$man ; \
 		done
 ifeq ($(ROOT),)
 		#
